### DISCUSSION TITLE
Any plan for LangChain integration?

### DISCUSSION THREAD

**Discussion by tigerinus** [openvinotoolkit/openvino#18575] [Q&A] [answered]
Would love to use model accelerated by Intel/OpenVINO in LangChain.

**Comment by devangaggarwal** [comment]
Thank you for this feedback and suggestion. We will definitely take this into account in our planning.

**Comment by nacartwright** [comment]
Ditto here too. I'm seeing customers being concerned about the token cost for using cloud models.

**Comment by yury-gorbachev** [comment]
This is good feature, I agree.
Is there any possibility to use this via HF adapter that they have and use optimum-intel for that I wonder?

**Comment by yury-gorbachev** [comment, marked as answer]
Meanwhile, you might want to check this: https://github.com/OpenVINO-dev-contest/langchain.openvino

We are considering upstreaming this, any feedback would be useful.

cc: @OpenVINO-dev-contest

**Comment by yury-gorbachev** [comment]
Just to keep this in the thread, update from @OpenVINO-dev-contest 

OpenVINO integration with LangChain is ready, including:
- LLM: https://python.langchain.com/docs/integrations/llms/openvino
- Embedding: https://python.langchain.com/docs/integrations/text_embedding/openvino
- Rerank：https://python.langchain.com/docs/integrations/document_transformers/openvino_rerank
